### PR TITLE
add hashjoin benchmark and fix dead link issue

### LINKED ISSUE #402: there is a dead link
It seems the Agaricus dataset has been moved to another place.
```
ERROR: 1 dead links found!
[✖] https://gust.dev/r/xgboost-agaricus → Status: 404

FILE: ./docs/get-started/xgboost-examples/prepare-package-data/preparation-python.md
[✓] https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/24.06.0/rapids-4-spark_2.12-24.06.0.jar → Status: 200
[/] /docs/get-started/xgboost-examples/building-sample-apps/python.md → Status: 0
[/] /docs/get-started/xgboost-examples/dataset/mortgage.md → Status: 0
[✓] https://www1.nyc.gov/site/tlc/about/tlc-trip-record-data.page → Status: 200
[✖] https://gust.dev/r/xgboost-agaricus → Status: 404
```

### PR DESCRIPTION
**Pull request by nvliyuan**
This pr is to :

- add hashjoin microbenchmark and update the latest performance
- fix #402 dead link issue

### CODE REVIEW

**Review by jlowe — COMMENTED** [review]
Need to update the [README](https://github.com/NVIDIA/spark-rapids-examples/blob/main/examples/SQL%2BDF-Examples/micro-benchmarks/README.md)

File: examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb [hunk 1]
```diff
@@ -541,18 +541,50 @@
...
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d8783a58-e3ee-4735-b8d0-6426e7cc16fe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### HashJoin\n",
```
Comment (viadea) [inline examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:551]: The "HashJoin" font is not bold as other titles?

File: examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb [hunk 2]
```diff
@@ -541,18 +541,50 @@
...
+    "This is a microbenchmark for a HashJoin. The query on GPU will be more than 10x times faster than CPU based on the cluster in the readme."
```
Comment (viadea) [inline examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:552]: Is it 10x or just 8.08x without enhanced configs on dataproc?

**Review by viadea — CHANGES_REQUESTED** [review]

File: examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb (hunk 2 above)
Comment (nvliyuan) [reply examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:552]: 8.08x is for dataproc, more than 10x(14x) is based on A100.

File: examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb (hunk 1 above)
Comment (nvliyuan) [reply examples/SQL+DF-Examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:549]: update the code cell to markdown cell

**Comment by nvliyuan** [comment]
> Need to update the [README](https://github.com/NVIDIA/spark-rapids-examples/blob/main/examples/SQL%2BDF-Examples/micro-benchmarks/README.md)

updated

File: examples/SQL+DF-Examples/micro-benchmarks/README.md [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
-The microbenchmark notebook in this repo uses four such queries in the chart shown below:
+The microbenchmark notebook in this repo uses five such queries in the chart shown below:
 
 - **Count Distinct**: a function used to estimate the number of unique page views or 
   unique customers visiting an e-commerce site.
 - **Window**: a critical operator necessary for preprocessing components in analyzing
   timestamped event data in marketing or financial industry.
 - **Intersect**: an operator used to remove duplicates in a dataframe.
 - **Cross-join**: A common use for a cross join is to obtain all combinations of items.
+- **Hash-join**: A common use for a hash join is to obtain all combinations of items.
```
Comment (jlowe) [inline examples/SQL+DF-Examples/micro-benchmarks/README.md:14]: Copy and paste error -- hash join is _not_ a way to generate all combinations of items.

Comment (nvliyuan) [reply examples/SQL+DF-Examples/micro-benchmarks/README.md]: updated

**Review by jlowe — APPROVED** [review]

**Pull request merged**